lib/spacesuit/auth_middleware.ex

```diff
@@ -29,6 +31,24 @@ defmodule Spacesuit.AuthMiddleware do
     end
   end
 
+  # Quick access function for the application settings for this middleware
+  def session_service do
+    Application.get_env(:spacesuit, :session_service)
```

Comment:
I did this for ease of testing so I could test different branches in the SUT but perhaps its not necessary if we simplify it as you've suggested. I'll update this patch with your suggestions and resubmit.